handshake_client.go

```diff
@@ -477,10 +529,20 @@ func (hs *clientHandshakeState) doFullHandshake() error {
 
 	keyAgreement := hs.suite.ka(c.vers)
 
+	// Set the public key used to verify the handshake.
+	pk := c.peerCertificates[0].PublicKey
```

Comment:
Quick question - I havn't looked enough in to code but I'm wondering why we can't reuse GetCertificate and VerifyPeerCertificate to get/validate credentials.
I know that those functions take/return "Certificate" object and you are using PublicKey and PrivateKey objects internally. But isn't it possible to wrap those keys (and other attributes) it in Certificate object? I think integration would be a bit easier then.

Reply:
I tried this originally but decided against. I don't exactly remember why it didn't work, but one reason is that choosing the DC requires knowing what TLS version has been negotiated. This can't be determined given only the parameters of `GetCertificate`.

Reply:
Seems to me that GetCertificate is a method of "ClientHelloInfo", which has a "Conn" which stores TLS version. But anyway, if you said you checked it, then it's probably not worth trying.

Reply:
Aw yeah you're right. But does crypto's Conn epxort the version?

Reply:
c.ver - it's used all over the code

Reply:
Yeah it doesn't export the version. I remember I looked at this a couple weeks ago.

Reply:
But if you're writing your own `GetCertificate`, you can't access ClientHelloInfo.Conn.vers.

Reply:
(Outside of crypto/tls)

Reply:
Oh. it's private field, but that's not  a problem at all. Simply adding public method will be both enough and actually useful

Reply:
We could add GetVersion() to tls.Conn. But the field in ClientHelloInfo is a net.Conn: Do you think this could be an issue?